app/frontend/javascript/custom_google_analytics_4_events.js

```diff
@@ -8,21 +8,39 @@ $(document).on('click', '*[data-analytics-category]', function(e) {
   // Do not call `gtag` unless the function is defined
   // in app/views/layouts/_google_analytics_4.html.erb .
   // (This in turn is controlled by ScihistDigicoll::Env.lookup(:google_analytics_4_tag_id).
+
+
   if (typeof gtag === 'function') {
 
+    // In certain cases we want to set event_value to the contents of a text input - one that
+    // might be different every time the link or button is clicked.
+
+    // Example use case: we want to set event_value to the query string in search-within-the-work search form.
+
+    // If we find a selector in data-analytics-value-css, we can look up the current value of the text input 
+    // using that selector, and then pass that string to GA as event_value.
+    var elementToLookUp =  e.currentTarget.getAttribute("data-analytics-value-css");
+    var eventValue = elementToLookUp ?  $(elementToLookUp)[0].value.replace(/[^a-zA-Z 0-9]+/g, '') : null;
```

Comment:
I guess we do want to capture the query ideally!  Although this wasn't part of the spec (are there privacy implications? Should we check with @Gabz73 ?)

Using a field pointing to a CSS selector is clever, I guess it works!

I feel like there was a reason I didn't use the GA `value` field before, like it was unsuitable for such things.. but also there's a whole new GA version we're using since then, let me look up the docs of GA "value" just to verify... I sort of feel like `value` had  to be an integer at least in old GA....

I'm having trouble finding docs. And the new GA is very different than the old GA!

But I'm wondering if we'd want to use a different mechanism in GA to capture queries, if we wanted to capture queries. Rather than metadata on a "event", there's a whole different mechanism I think...  but I'm having trouble finding that too, GA docs are terrible!

I wonder if we watn to bail on tracking queries unless that's been specifically asked for by stakeholders?  Alternatively, have you actually verified this works to capture search terms *in a way that can be usably accessed in GA console*??

Reply:
It actually looks to me like maybe we can put _arbitrary_ metadata on the event in GA4 now, when before we were limited? So that is positive for this working, altough I still want to see it verified in GA Search Console that this new value can be accessed usefully, before committing code to record it!

And if we can call the metadata lable anything we want, should we actually call it "search_terms" or something?

Not totally sure about any of this!

https://developers.google.com/tag-platform/gtagjs/reference#event